math/src/fft/tests.rs

```diff
@@ -20,40 +20,40 @@ fn fft_in_place() {
     let n = 4;
     let mut p = rand_vector(n);
     let domain = build_domain(n);
-    let expected = polynom::eval_many(&p, &domain);
+    let expected = polynom::eval_many(p.as_mut_slice(), &domain);
     let twiddles = super::get_twiddles::<BaseElement>(n);
-    super::serial::fft_in_place(&mut p, &twiddles, 1, 1, 0);
-    super::permute(&mut p);
+    super::serial::fft_in_place(p.as_mut_slice(), &twiddles, 1, 1, 0);
+    super::permute(p.as_mut_slice());
```

Comment:
is there a reason not to revert these changes?

Reply:
We have to  pass a slice as an argument in `fft_in_place`. Rest all I think can be reverted back.